frontend/src/citizen-frontend/messages/MessagesPage.tsx

```diff
@@ -79,15 +85,21 @@ export default React.memo(function MessagesPage() {
     setSelectedThread(params.threadId)
   }, [setSelectedThread, params.threadId])
 
+  const initThreadFocusRef = useRef<() => void>()
+
   const selectThread = useCallback(
     (threadId: UUID | undefined) => {
       if (!threadId) {
         navigate('/messages')
       } else {
-        navigate(`/messages/${threadId}`)
+        if (params.threadId !== threadId) {
+          navigate(`/messages/${threadId}`)
+        } else {
+          initThreadFocusRef.current?.()
```

Comment:
sitten täällä kutsutaan titlen focus jos thread ei vaihdu
```suggestion
          threadTitleRef.current?.focus()
```